pkg/retry/failure_retryer.go

```diff
@@ -284,3 +286,13 @@ func (r *failureRetryer) Dump() map[string]interface{} {
 	}
 	return dm
 }
+
+func (r *failureRetryer) setSpecifiedResultRetryIfNeeded(rr *ShouldResultRetry) {
+	if rr != nil {
+		r.specifiedResultRetry = rr
+	}
+	if r.policy != nil && r.policy.ShouldResultRetry == nil {
+		// the ShouldResultRetry priority of that config inside FailurePolicy is higher
```

Comment:
The priority of FailurePolicy.ShouldResultRetry is higher, so only update it when it's nil